spot-client/src/spot-remote/app-state/actions.js

```diff
@@ -86,40 +86,33 @@ export function connectToSpotTV(joinCode, shareMode) {
         });
 
         const state = getState();
-        const { joinCodeServiceUrl } = getSpotServicesConfig(getState());
-
-        return remoteControlClient.exchangeCode(joinCode, { joinCodeServiceUrl })
-            .then(roomInfo => remoteControlClient.connect({
-                autoReconnect: true,
-                roomInfo,
-                serverConfig: getRemoteControlServerConfig(state)
-            }))
-            .then(() => {
-                dispatch({
-                    type: SPOT_REMOTE_JOIN_CODE_VALID,
-                    joinCode,
-                    shareMode
-                });
-            })
-            .catch(error => {
-                dispatch({
-                    type: SPOT_REMOTE_JOIN_CODE_INVALID,
-                    joinCode,
-                    shareMode
-                });
-
-                // In the wrong password case return back to join code entry.
-                if (error === 'not-authorized') {
-                    dispatch(addNotification('error', 'Invalid share key'));
-                } else if (error === 'unrecoverable-error') {
-                    dispatch(addNotification('error', 'Something went wrong'));
-                }
-
-                _onDisconnected(error, {
```

Comment:
You don't need this call to _onDisconnected anymore?

Reply:
oh did it slip through in some of the commits ? No, it's not needed after your change to call disconnected when the connect promise is rejected. Will double check.

Reply:
Okay, so it changed when the exchangeCode call was moved to be part of the connect. Because when it's part of connect the RCS will fire disconnected when it fails. Prior to that it needed to be called no `exchangeCode` rejection.